pymongo/mongo_client.py

```diff
@@ -1408,12 +1408,17 @@ def is_retrying():
                 if retryable_error:
                     session._unpin()
                 if not retryable_error or (is_retrying() and not multiple_retries):
-                    raise
+                    if exc.has_error_label("NoWritesPerformed"):
+                        assert last_error is not None
```

Comment:
I assume this is here for the type checker.  I think we should be more conservative and check `if exc.has_error_label("NoWritesPerformed") and last_error is not None`